internal/workload/workload.go

```diff
@@ -195,5 +225,83 @@ func (w *WorkloadManager) toPodYaml(workload *models.Workload) ([]byte, error) {
 	}
 	pod.Kind = "Pod"
 	pod.Name = workload.Name
-	return yaml.Marshal(pod)
+	return &pod, nil
+}
+
+func (ww workloadWrapper) Init() error {
+	return ww.netfilter.AddTable(nftTableName)
+}
+
+func (ww workloadWrapper) List() ([]api2.WorkloadInfo, error) {
+	return ww.workloads.List()
+}
+
+func (ww workloadWrapper) Remove(workloadName string) error {
+	if err := ww.workloads.Remove(workloadName); err != nil {
+		return err
+	}
+	if err := ww.netfilter.DeleteChain(nftTableName, workloadName); err != nil {
+		log.Errorf("Failed to delete chain %[1]s from %s table for workload %[1]s", workloadName, nftTableName)
+	}
+	return nil
+}
+
+func (ww workloadWrapper) Run(workload *v1.Pod, manifestPath string) error {
+	if err := ww.applyNetworkConfiguration(workload); err != nil {
+		return err
+	}
+	if err := ww.workloads.Run(manifestPath); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ww workloadWrapper) applyNetworkConfiguration(workload *v1.Pod) error {
+	hostPorts, err := getHostPorts(workload)
+	if err != nil {
+		return err
+	}
+	if len(hostPorts) == 0 {
+		return nil
+	}
+	//TODO: should verify if chain doesn't exist
+	if err := ww.netfilter.AddChain(nftTableName, workload.Name); err != nil {
+		// should we return nil and only log the error ?
+		return err
+	}
+
+	// for workloads, a port will be opened for the pod based on hostPort
+	for _, p := range hostPorts {
+		rule := fmt.Sprintf("tcp dport %d ct state new,established counter accept", p)
+		if err := ww.netfilter.AddRule(nftTableName, workload.Name, rule); err != nil {
+			// should we return nil and only log the error ?
```

Comment:
In my opinion we should behave in similar way as for `addChain`. Log and return an error.